File: app/src/main/java/com/coinninja/coinkeeper/di/builder/AndroidBroadcastReceiverBuilder.java
package com.coinninja.coinkeeper.di.builder;

import com.coinninja.coinkeeper.receiver.ApplicationStartedReceiver;
import com.coinninja.coinkeeper.receiver.AuthenticationCompleteReceiver;
import com.coinninja.coinkeeper.receiver.DeviceRebootBootCompletedReceiver;
import com.coinninja.coinkeeper.receiver.StartupCompleteReceiver;
import com.coinninja.coinkeeper.receiver.WalletCreatedBroadCastReceiver;
import com.coinninja.coinkeeper.receiver.WalletRegistrationCompleteReceiver;
import com.coinninja.coinkeeper.receiver.WalletSyncCompletedReceiver;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class AndroidBroadcastReceiverBuilder {

    @ContributesAndroidInjector
    abstract ApplicationStartedReceiver applicationStartedReceiver();

    @ContributesAndroidInjector
    abstract StartupCompleteReceiver startupCompleteReceiver();

    @ContributesAndroidInjector
    abstract WalletCreatedBroadCastReceiver walletCreatedBroadCastReceiver();

    @ContributesAndroidInjector
    abstract WalletRegistrationCompleteReceiver walletRegistrationCompleteReceiver();

    @ContributesAndroidInjector
    abstract WalletSyncCompletedReceiver walletSyncCompletedReceiver();

    @ContributesAndroidInjector
    abstract DeviceRebootBootCompletedReceiver deviceRebootBootCompletedReceiver();

    @ContributesAndroidInjector
    abstract AuthenticationCompleteReceiver authenticationCompleteReceiver();
}

File: app/src/main/java/com/coinninja/coinkeeper/di/builder/AndroidBroadcastReceiverBuilder.kt
package com.coinninja.coinkeeper.di.builder

import com.coinninja.coinkeeper.receiver.*
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class AndroidBroadcastReceiverBuilder {

    @ContributesAndroidInjector
    internal abstract fun applicationStartedReceiver(): ApplicationStartedReceiver

    @ContributesAndroidInjector
    internal abstract fun startupCompleteReceiver(): StartupCompleteReceiver

    @ContributesAndroidInjector
    internal abstract fun walletCreatedBroadCastReceiver(): WalletCreatedBroadCastReceiver

    @ContributesAndroidInjector
    internal abstract fun walletRegistrationCompleteReceiver(): WalletRegistrationCompleteReceiver

    @ContributesAndroidInjector
    internal abstract fun walletSyncCompletedReceiver(): WalletSyncCompletedReceiver

    @ContributesAndroidInjector
    internal abstract fun deviceRebootBootCompletedReceiver(): DeviceRebootBootCompletedReceiver

    @ContributesAndroidInjector
    internal abstract fun authenticationCompleteReceiver(): AuthenticationCompleteReceiver
}
